test/jdk/java/awt/TextArea/SelectionVisible/SelectionVisible.java

```diff
@@ -21,42 +21,64 @@
  * questions.
  */
 
-
-import java.applet.Applet;
-import java.awt.BorderLayout;
-import java.awt.Dimension;
-import java.awt.FlowLayout;
-import java.awt.Panel;
+import java.awt.Frame;
 import java.awt.TextArea;
+import java.lang.reflect.InvocationTargetException;
+
+import static java.awt.EventQueue.invokeAndWait;
+
+/*
+  @test
+  @key headful
+  @bug 4082144 7150100
+  @summary  Ensures that TextArea.select() works when called
+            before setVisible()
+  @run main SelectionVisible
+*/
 
-public final class SelectionVisible extends Applet {
+public class SelectionVisible {
 
-    private TextArea ta;
+    private static TextArea ta;
+    private static Frame frame;
 
-    @Override
-    public void init() {
+    public static void createTestUI() {
+        frame = new Frame("Test 4082144 7150100");
         ta = new TextArea(4, 20);
         ta.setText("01234\n56789");
         ta.select(3, 9);
 
-        final TextArea instruction = new TextArea("INSTRUCTIONS:\n"
-                                                 + "The text 34567 should be selected in the TextArea.\n"
-                                                 + "If this is what you observe, then the test passes.\n"
-                                                 + "Otherwise, the test fails.", 40, 5,
-                                         TextArea.SCROLLBARS_NONE);
-        instruction.setEditable(false);
-        instruction.setPreferredSize(new Dimension(300, 70));
-        final Panel panel = new Panel();
-        panel.setLayout(new FlowLayout());
-        panel.add(ta);
-        setLayout(new BorderLayout());
-        add(instruction, BorderLayout.CENTER);
-        add(panel, BorderLayout.PAGE_END);
-    }
+        frame.add(ta);
+        frame.setAlwaysOnTop(true);
+        frame.setLocationRelativeTo(null);
+        frame.pack();
+        frame.setVisible(true);
 
-    @Override
-    public void start() {
-        setVisible(true);
         ta.requestFocus();
     }
+
+    public static void test() {
+        String selectedText = ta.getSelectedText();
+        System.out.println("selectedText : " + selectedText);
+        if (!selectedText.equals("34\n567")) {
+            disposeFrame();
+            throw new RuntimeException("Expected '34\n567' to be " +
+                    "selected text, but got " + selectedText);
```

Comment:
It probably doesn't matter but I'd like to see disposeFrame called with invokeAndWait
And you could restructure this to be simpler
invokeAndWait(::disposeFrame)
if (!selectedText ...) {
} else {
System.out.println("test passed");
}

and then you only need one call site for the dispose